OpenSim/Common/Test/testComponentInterface.cpp

```diff
@@ -1889,14 +1895,9 @@ void testSingleValueInputConnecteeSerialization() {
     // Deserialize.
     {
         // Single-value connectee cannot have multiple connectee_names.
-        // TODO Would ideally check for an exception, but we only emit a warning
-        // for now. This is because the way we determine if multiple
-        // connectee names were specified is by looking for spaces, but old
-        // models used have spaces in their names.
-        // SimTK_TEST_MUST_THROW_EXC(
-        //     TheWorld world(modelFileNameMultipleValues),
-        //     OpenSim::Exception);
-        TheWorld world(modelFileNameMultipleValues);
+        SimTK_TEST_MUST_THROW_EXC(
+             TheWorld world(modelFileNameMultipleValues),
+             OpenSim::Exception);
```

Comment:
I'm happy that this TODO can be resolved now, but I'm not sure which exception is being caught; it can't be the desired one. The code that generates the warning is in [ComponentSocket.h](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/ComponentSocket.h#L248), and you can see a commented-out exception there. That's the exception that is supposed to be caught here. Could you also resolve the TODO in ComponentSocket.h and ensure that exception is the one that's caught here?

Reply:
@chrisdembia I beefed `ComponentSocket::checkConnecteeNameProperty() to check for invalid characters using `ComponentPath::isLegalPathElement()` and I have it throw.  It is indeed that exception that is being caught here, as intended.

Reply:
Okay great.